octobot_trading/personal_data/orders/order_util.py

```diff
@@ -215,3 +216,31 @@ async def create_as_chained_order(order):
             )
     finally:
         order.created = True
+
+
+@contextlib.asynccontextmanager
+async def ensure_orders_relevancy(order=None, position=None):
+    exchange_manager = order.exchange_manager if position is None else position.exchange_manager
+    # part used in futures trading only
+    if exchange_manager.exchange_personal_data.positions_manager.positions:
+        position = position or exchange_manager.exchange_personal_data.positions_manager.get_order_position(order)
+        pre_update_position_side = position.side
+        pre_update_position_size = position.size
+        yield
+        if position.side != pre_update_position_side or \
+           (position.size is constants.ZERO and pre_update_position_size is not constants.ZERO):
+            # when position side is changing or size is going back to 0, associated reduce only orders must be closed
+            await _update_reduce_only_orders_on_position_reset(exchange_manager, position.symbol)
+    else:
+        # as a context manager, yield is mandatory
+        yield
+
+
+async def _update_reduce_only_orders_on_position_reset(exchange_manager, symbol):
```

Comment:
This is not really an "update" but a "close" or a "cancel". I would prefer to name this method `_close_reduce_only...`